### PR TITLE
Final tweaks to vertically oriented text / labels

### PR DESCRIPTION
**Pull request by nirvn**
## Description

@nyalldawson , curved placement isn't compatible with non horizontal orientation for a few reasons. When it comes to rotation-based orientation, that mode can result is a mixture of rotated and non-rotated characters for the same label (bad!); for the vertical orientation mode, since we break strings into graphemes, it ends up rotating 90 degrees along the curves, which is also nonsensical.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `Fixes #11111` at the bottom of the commit message
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit
- [x] I have evaluated whether it is appropriate for this PR to be backported, backport requests are left as label or comment

### CODE REVIEW

**Comment by nirvn** [comment]
The second commit in this PR (and hopefully the final one) fixes rendering of a number of characters (parenthesis, ellipses, etc.) in vertically oriented text by substituting the characters with the vertical presentation counterparts.

Screenshot from 2019-08-16 11-04-23

File: src/core/qgstextrenderer.cpp [hunk 1]
```diff
@@ -3550,8 +3550,28 @@ void QgsTextRenderer::drawTextInternal( TextPart drawType,
 
       bool adjustForAlignment = alignment != AlignLeft && ( mode != Label || textLines.size() > 1 );
 
-      const auto constTextLines = textLines;
-      for ( const QString &line : constTextLines )
+      // apply some character replacement to draw symbols in vertical presentation
```
Comment (nyalldawson) [inline src/core/qgstextrenderer.cpp]: Might be nice to move this to QgsStringUtils

**Comment by nyalldawson** [comment]
> The second commit in this PR (and hopefully the final one) fixes rendering of a number of characters (parenthesis, ellipses, etc.) in vertically oriented text by substituting the characters with the vertical presentation counterparts.

Nice! I was wondering if that would be an issue :)

File: src/core/qgstextrenderer.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/qgstextrenderer.cpp]: @nyalldawson , good idea, will do.

**Pull request merged**